core/src/test/java/org/apache/iceberg/TestSnapshotProducer.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+public class TestSnapshotProducer {
+  @Test
+  public void testManifestFileGroupSize() {
+    assertEquals(
+        1_000,
+        SnapshotProducer.manifestFileGroupSize(4 /* parallelism */, 1_000 /* file count */),
+        "Must return file count when it is small");
+
+    assertEquals(
+        10_000,
+        SnapshotProducer.manifestFileGroupSize(4 /* parallelism */, 10_000 /* file count */),
+        "Must return file count when it matches the min group size");
+
+    assertEquals(
+        10_001,
+        SnapshotProducer.manifestFileGroupSize(4 /* parallelism */, 10_001 /* file count */),
+        "Must return file count when it is slightly above min group size");
+
+    assertEquals(
+        12_500,
+        SnapshotProducer.manifestFileGroupSize(4 /* parallelism */, 12_500 /* file count */),
+        "Must return file count when remainder is < 50% of min group size");
+
+    assertEquals(
+        7_500,
+        SnapshotProducer.manifestFileGroupSize(4 /* parallelism */, 15_000 /* file count */),
+        "Must split into 2 groups when remainder is >= 50% of min group size");
+
+    assertEquals(
+        16_667,
+        SnapshotProducer.manifestFileGroupSize(3 /* parallelism */, 50_000 /* file count */),
+        "Must split into 3 groups when file count is large");
```

Comment:
this test is essentially the same as the one below. note that min group size is 10_000.